src/v2/view-builder/utils/AuthenticatorUtil.js

```diff
@@ -30,154 +30,163 @@ const getAuthenticatorData = function(authenticator, isVerifyAuthenticator) {
+      Object.assign(authenticatorData, {
+        description: isVerifyAuthenticator
+          ? ''
+          : loc('oie.rsa.authenticator.description', 'login'),
+        iconClassName: 'mfa-rsa',
+        buttonDataSeAttr: getButtonDataSeAttr(authenticator),
+      });
+      break;
+
+    case AUTHENTICATOR_KEY.DUO:
+      Object.assign(authenticatorData, {
+        description: isVerifyAuthenticator
+          ? ''
+          : loc('oie.duo.authenticator.description', 'login'),
+        iconClassName: 'mfa-duo',
+        buttonDataSeAttr: getButtonDataSeAttr(authenticator),
+      });
+      break;
+
+    case AUTHENTICATOR_KEY.IDP: {
+      const idpName =  authenticator.relatesTo?.displayName;
+      Object.assign(authenticatorData, {
+        description: isVerifyAuthenticator
+          ? ''
+          : loc('oie.idp.authenticator.description', 'login', [idpName]),
+        iconClassName: 'mfa-custom-factor',
+        buttonDataSeAttr: getButtonDataSeAttr(authenticator),
+      });
+      break;
+    }
+    case AUTHENTICATOR_KEY.CUSTOM_OTP: {
+      Object.assign(authenticatorData, {
+        description: isVerifyAuthenticator
+          ? ''
+          : loc('oie.custom_otp.description', 'login'),
+        iconClassName: 'mfa-hotp',
+        buttonDataSeAttr: getButtonDataSeAttr(authenticator),
+      });
+      break;
+    }
 
-  case AUTHENTICATOR_KEY.SYMANTEC_VIP: {
-    const appName =  authenticator.relatesTo?.displayName;
-    Object.assign(authenticatorData, {
-      description: isVerifyAuthenticator
-        ? ''
-        : loc('oie.symantecVip.authenticator.description', 'login', [appName]),
-      iconClassName: 'mfa-symantec',
-      buttonDataSeAttr: getButtonDataSeAttr(authenticator),
-    });
-    break;
-  }
+    case AUTHENTICATOR_KEY.SYMANTEC_VIP: {
+      const appName =  authenticator.relatesTo?.displayName;
+      Object.assign(authenticatorData, {
+        description: isVerifyAuthenticator
+          ? ''
+          : loc('oie.symantecVip.authenticator.description', 'login', [appName]),
+        iconClassName: 'mfa-symantec',
+        buttonDataSeAttr: getButtonDataSeAttr(authenticator),
+      });
+      break;
+    }
 
-  case AUTHENTICATOR_KEY.YUBIKEY: {
-    Object.assign(authenticatorData, {
-      description: isVerifyAuthenticator
-        ? ''
-        : loc('oie.yubikey.authenticator.description', 'login'),
-      iconClassName: 'mfa-yubikey',
-      buttonDataSeAttr: getButtonDataSeAttr(authenticator),
-    });
-    break;
-  }
+    case AUTHENTICATOR_KEY.YUBIKEY: {
+      Object.assign(authenticatorData, {
+        description: isVerifyAuthenticator
+          ? ''
+          : loc('oie.yubikey.authenticator.description', 'login'),
+        iconClassName: 'mfa-yubikey',
+        buttonDataSeAttr: getButtonDataSeAttr(authenticator),
+      });
+      break;
+    }
+
+    case AUTHENTICATOR_KEY.CUSTOM_APP: {
+      Object.assign(authenticatorData, {
+        description: '',
```

Comment:
nit: should we add a `TODO` here for when enroll is available? All the other authenticators have `description` value based on `isVerifyAuthenticator`, that is why I am asking

Reply:
@mauriciocastillosilva-okta  will update this when I work on enroll.